File: regressproof/fixtures/swift-js/tracked/current/scripts/typecheck.swift
import Foundation

let source = try String(contentsOfFile: "src/App.swift", encoding: .utf8)

if source.contains("TYPE_FAIL") {
    FileHandle.standardError.write(Data("typecheck failed in src/App.swift: found TYPE_FAIL marker\n".utf8))
    exit(1)
}

print("typecheck passed")

File: regressproof/fixtures/simple-js/tracked/baseline/scripts/typecheck.js
const fs = require("node:fs");

const source = fs.readFileSync("src/app.js", "utf8");

if (source.includes("TYPE_FAIL")) {
  console.error("typecheck failed: found TYPE_FAIL marker");
  process.exit(1);
}

console.log("typecheck passed");
